[docs] EAS Build - workaround for npm cache for yarn v1 (#12062)
expo/expo: docs/pages/build-reference/infrastructure.md

- Android workers run on Kubernetes in an isolated environment
  - Every build gets its own container running on a dedicated Kubernetes node
  - Build resources: 4 CPU, 16 GB RAM (14 GB after k8s overhead)

> - NPM cache deployed with Kubernetes (works with npm and yarn v2, for compatibility with yarn v1 checkout [workaround](how-tos/#using-npm-cache-with-yarn-v1))

Reviewer: ```suggestion
- NPM cache deployed with Kubernetes (works with npm and yarn v2, and yarn v1 works with a [workaround](how-tos/#using-npm-cache-with-yarn-v1))
```

  - Every build gets its own fresh macOS VM
  - Hardware: Intel(R) Xeon(R) CPU E5-2697 (12 core/24 threads), 64 GB RAM
  - Build resource limits: 6 cores, 8 GB RAM

> - NPM cache (works with npm and yarn v2, for compatibility with yarn v1 checkout [workaround](how-tos/#using-npm-cache-with-yarn-v1))

Reviewer: ```suggestion
- NPM cache (works with npm and yarn v2, and yarn v1 works with a [workaround](how-tos/#using-npm-cache-with-yarn-v1))
```
